lib/mongo/cluster.rb

```diff
@@ -66,7 +66,7 @@ def ==(other)
     # @since 2.0.0
     def add(host)
       address = Address.new(host)
-      unless addresses.include?(address)
+      unless addresses.include?(address) || (@topology.single? && address.seed != @topology.seed)
```

Comment:
@durran This line can probably be improved, let me know if you have any other ideas.